enumerable_methods.rb

```diff
@@ -0,0 +1,155 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      if is_a? Hash
+        yield(keys[i], self[keys[i]])
+        return self if i == size - 1
+      elsif is_a? Array
+        yield self[i]
+        return self if i == size - 1
+      elsif is_a? Range
+        yield to_a[i]
+        return self if i == size - 1
+      end
+      i += 1
+    end
+  end
```

Comment:
`my_each` identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))  
- calls the given block once for each element in self. But a case is failing. Check out the code below. You should see get same output twice.
```ruby
hash = { a: 1, b: 2, c: 3, d: 4, e: 5 }
my_each_output = ''
hash.my_each { |num| my_each_output += num.to_s }
p my_each_output
my_each_output = ''
hash.each { |num| my_each_output += num.to_s }
p my_each_output
```
Well, there is a way to resolve it, easily and it can refactor your code a lot more.
```ruby
def my_each
    return to_enum(:my_each) unless block_given?
    
    converted_array = self.to_a
    i = 0
    while i < converted_array.size
      yield(converted_array[i])
      i += 1
    end
    self
end
```